tests/stats_test.cpp

```diff
@@ -0,0 +1,415 @@
+}
+
+/*
+ * Top statistics handler must exist, if "top" section in "monitor" section in config
+ * exists. Node's configuration happened in configure_node() function above.
+ */
+static void test_top_statistics_existence()
+{
+	dnet_node *node = global_data->nodes[0].get_native();
+	auto monitor = ioremap::monitor::get_monitor(node);
+	BOOST_CHECK(monitor != nullptr);
+
+	auto top_stats = monitor->get_top_stats();
+	BOOST_CHECK(top_stats != nullptr);
+}
+
+/****************
+ Test event_stats
+ ****************/
+typedef ioremap::monitor::event_stats<test_event> stats_t;
+
+static void test_event_stats_boundary_conditions()
+{
+	const size_t default_size = 100;
+	const time_t default_time = time(nullptr);
+	const time_t expire_time = default_time + PERIOD_IN_SECONDS + 1;
+	stats_t stats(EVENTS_SIZE, PERIOD_IN_SECONDS);
+	std::vector<test_event> result;
+
+	// get_top returns empty list, if there was no added events
+	stats.get_top(TOP_LENGTH, default_time, result);
+	BOOST_CHECK(result.empty());
+
+	// get_top returns no more events than was requested
+	for(int i = 0; i < 2 * TOP_LENGTH; ++i) {
+		test_event e{std::to_string(static_cast<long long>(i)), default_size, 1., default_time};
+		stats.add_event(e, e.get_time());
+	}
+
+	stats.get_top(TOP_LENGTH, default_time, result);
+	BOOST_REQUIRE_EQUAL(result.size(), TOP_LENGTH);
+
+	// get_top returns no more events than EVENTS_LIMIT
+	for(int i = 0; i < 4 * EVENTS_LIMIT; ++i) {
+		test_event e{std::to_string(static_cast<long long>(rand())), default_size, 1., default_time};
+		stats.add_event(e, e.get_time());
+	}
+
+	result.clear();
+	stats.get_top(4*EVENTS_LIMIT, default_time, result);
+	BOOST_REQUIRE_EQUAL(result.size(), EVENTS_LIMIT);
+
+	// all events has been expired since expire_time elapsed
+	result.clear();
+	stats.get_top(TOP_LENGTH, expire_time, result);
+	BOOST_CHECK(result.empty());
+
+	// get_top should not return more events, than were added
+	const int few_events = 5;
+	BOOST_CHECK(few_events < TOP_LENGTH);
+
+	for(int i = 0; i < few_events; ++i) {
+		test_event e{std::to_string(static_cast<long long>(i)), default_size, 1., default_time};
+		stats.add_event(e, e.get_time());
+	}
+
+	stats.get_top(TOP_LENGTH, default_time, result);
+	BOOST_REQUIRE_EQUAL(result.size(), few_events);
+}
+
+template<typename container_t>
+static size_t events_symmetric_diff(container_t &fst, container_t &snd)
+{
+	std::sort(fst.begin(), fst.end(), test_event::key_compare_event);
+	std::sort(snd.begin(), snd.end(), test_event::key_compare_event);
+
+	std::vector<test_event> diff(fst.size() + snd.size());
+	auto it = std::set_symmetric_difference(fst.begin(), fst.end(),
+					        snd.begin(), snd.end(),
+					        diff.begin(), test_event::key_compare_event);
+	return std::distance(diff.begin(), it);
+}
+
+/*
+ * Check max-heap property of event_stats. All events inserted at the same time.
+ * All events describe access to different keys.
+ */
+static void test_event_stats_no_time_dependency()
```

Comment:
This needs more details - what this test is supposed to check, how you do this, what results are going to be received, how will you interpret them, how will you check that they are correct or wrong and so on